ReactSkia/views/common/RSkDrawUtils.cpp

```diff
@@ -76,13 +81,112 @@ void setColor(SharedColor Color,SkPaint *paint)
 }
 bool isDrawVisible(SharedColor Color,Float thickness=1.0)
 {
-    return (Color !=clearColor() && thickness > 0.0)? true:false;
+    return ((colorComponentsFromColor(Color).alpha != 0) && (thickness != 0.0))? true:false;
+}
+
+FrameType getFrameBorderType(BorderMetrics borderProps,SharedColor backgroundColor)
+{
+    FrameType frameType=InvisibleFrame;
+    if( !borderProps.borderWidths.isUniform() ){
+        frameType=DiscretePath;
+    } else if( borderProps.borderWidths.left ==0) {
+        //Width confirmed to be uniform, make sure it is visible
+        frameType=InvisibleFrame;
+    } else {
+       /*All the sides confirmed to have uniform visble Thickness.
+         confirming visible color on ALL sides*/
+        if(borderProps.borderColors.isUniform() && isDrawVisible(borderProps.borderColors.left)){
+            frameType=MonoChromeStrokedRect;
+        } else if(!borderProps.borderColors.isUniform()) {
+        /* Border Color  differs for each Side.So make sure all the side are Visible*/
+            if(isDrawVisible(borderProps.borderColors.left,borderProps.borderWidths.left) &&
```

Comment:
With teh new approach, just need to validate colors and not width. SO can make a seperate inline Api to validate color